### PR TITLE
Sparrow dom/origin wallet social native again

### PR DESCRIPTION
**Pull request by sparrowDom**
### Description:

Use deep links to intercept and:
 - open origin's twitter profile in twitter app (Android)
 - start a tweet in twitter app (Android)
 - open Origin's facebook page (Android, IOS)

This PR is a better alternative to: https://github.com/OriginProtocol/origin/pull/2928
### Checklist:

- [x] Test your work and double-check to confirm that you didn't break anything
- [ ] [Wrap any new text/strings with `fbt`](https://github.com/OriginProtocol/origin/tree/master/dapps/marketplace/translation#wrapping-text) for translation
- [ ] If there are any new/changed strings to translate, [`npm run translate`](https://github.com/OriginProtocol/origin/tree/master/dapps/marketplace/translation#integrating-translations)
- [ ] Update any relevant READMEs and [docs](https://github.com/OriginProtocol/origin/tree/master/docs)

### CODE REVIEW

File: mobile/src/screens/marketplace.js [hunk 1]
```diff
@@ -564,9 +565,71 @@ class MarketplaceScreen extends Component {
...
+
+  checkForShareNativeDialogInterception = async url => {
+    // natively tweet if possible on Android
+    await this._openDeepLinkUrlAttempt(
+      () =>
+        url.hostname === 'twitter.com' &&
+        url.pathname === '/intent/tweet' &&
+        Platform.OS === 'android',
+      () =>
+        `twitter://post?message=${encodeURIComponent(
```
Comment (tomlinton) [inline mobile/src/screens/marketplace.js]: What happens with these kind of links if the user doesn't have Twitter or Facebook? Do they just fail silently?

File: mobile/src/screens/marketplace.js [hunk 2]
```diff
@@ -564,9 +565,71 @@ class MarketplaceScreen extends Component {
...
+  _openDeepLinkUrlAttempt = async (
+    interceptUrlPredicate,
+    makeUrl,
+    timeControlVariableName
+  ) => {
+    // non interceptable url
+    if (!interceptUrlPredicate()) return
+
+    const url = makeUrl()
+    if (Linking.canOpenURL(url)) {
```
Comment (tomlinton) [inline mobile/src/screens/marketplace.js]: Ahh I guess this isn't true if they don't have Facebook/Twitter?

Comment (sparrowDom) [reply mobile/src/screens/marketplace.js]: yeah it just opens the thing in the native browser as it does right now.

**Comment by tomlinton** [comment]
Looks good to me, have you tested it out? Happy to spin it up if you haven't.

**Comment by sparrowDom** [comment]
Thanks @tomlinton . Yeah have tested this on Android and iPhone

**Pull request merged**